src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,149 @@
+     */
+    public String add(Task t) {
+        tasks.add(t);
+
+        return "Got it. BMO has added this task:\n      " + t + count();
+    }
+
+    /**
+     * Returns a String to represent the current list of Tasks
+     *
+     * @return Current list of Tasks
+     */
+    public String list() {
+        String tempStr = "Here are the tasks in your list:\n    ";
+        for (int i = 0; i < tasks.size(); i++) {
+            if (i != 0) {
+                tempStr += "\n    ";
+            }
+            tempStr += (i + 1) + ". " + tasks.get(i);
+        }
+
+        return tempStr;
+    }
+
+    /**
+     * Marks the Task according to the index provided and returns a message for successfully marking the Task
+     *
+     * @param idx index of the Task to be marked
+     * @return Message for successfully marking the Task
+     * @throws DukeException If the index specified is out of bounds OR If the TaskList is empty
+     */
+    public String mark(int idx) throws DukeException {
+                
+        if (idx < 1 || idx > tasks.size()) {
+            throw new DukeIdxOOBException();
+        }
+        
+        tasks.get(idx - 1).mark();
+        
+        return "Awesome! :D Another task done:\n      " + tasks.get(idx - 1);
+
+    }
+
+    /**
+     * Unmarks the Task according to the index provided and returns a message for successfully unmarking the Task
+     *
+     * @param idx index of the Task to be unmarked
+     * @return Message for successfully unmarking the Task
+     * @throws DukeException If the index specified is out of bounds OR If the TaskList is empty
+     */
+    public String unmark(int idx) throws DukeException {
+                
+        if (idx < 1 || idx > tasks.size()) {
+            throw new DukeIdxOOBException();
+        }
+        
+        tasks.get(idx - 1).unmark();
+        
+        return "T_T BMO can't believe you lied to me before.. I will unmark this task:\n      " + tasks.get(idx - 1);
+
+    }
+
+    /**
+     * Deletes the Task according to the index provided and returns a message for successfully deleting the Task
+     *
+     * @param idx index of the Task to be deleted
+     * @return Message for successfully deleting the Task
+     * @throws DukeException If the index specified is out of bounds OR If the TaskList is empty
+     */
+    public String delete(int idx) throws DukeException {
+
+        if (tasks.size() == 0) {
+            throw new DukeException("BMO don't think you have anything to delete :\\");
+        }
+
+        if (idx < 1 || idx > tasks.size()) {
+            throw new DukeIdxOOBException();
+        }
+
+        return "BMO deleted the task. Yay!:\n      " + tasks.remove(idx - 1) + count();
+
+    }
+
+    /**
+     * Returns a string that describes the number of tasks in the list
+     *
+     * @return String that describes the number of tasks in the TaskList
+     */
+    public String count() {
+        String isSingular = "s";
```

Comment:
Why does the string variable had the prefix "is", it might be confusing for others as this prefix is reserved for boolean variables. It might be better to rename this variable :)